src/storage/heap_file.c

```diff
@@ -24959,3 +24479,38 @@ heap_scan_cache_allocate_recdes_data (THREAD_ENTRY * thread_p, HEAP_SCANCACHE *
 
   return NO_ERROR;
 }
+
+/*
+ * heap_get_class_record () - Retrieves class objects only
+ * 
+ * return SCAN_CODE: S_SUCCESS or error
+ * thread_p (in)   : Thread entry.
+ * class_oid (in)  : Class object identifier.
+ * recdes_p (out)  : Record descriptor.
+ * scan_cache (in) : Scan cache.
+ * ispeeking (in)  : PEEK or COPY
+ */
+SCAN_CODE 
+heap_get_class_record (THREAD_ENTRY * thread_p, OID *class_oid, RECDES * recdes_p, HEAP_SCANCACHE * scan_cache, int ispeeking)
+{
+  HEAP_GET_CONTEXT context;
+  OID root_oid = *oid_Root_class_oid;
+  SCAN_CODE scan;
+
+#if !defined(NDEBUG)
+  /* for debugging set root_oid NULL and check afterwards if it really is root oid */
+  OID_SET_NULL (&root_oid);
+#endif // !NDEBUG
+
+  heap_init_get_context (thread_p, &context, class_oid, &root_oid, recdes_p, scan_cache, PEEK, NULL_CHN);
+  
```

Comment:
@andrei14vl 

I think `PEEK` should be `ispeeking`. 
Some callers of the function expects `COPY`.